src/GCDDisplay.cs

```diff
@@ -0,0 +1,299 @@
+            }
+            ui.DrawCircSegment(0f, Math.Min(gcdTime / gcdTotal, 1f), 20f * ui.Scale, conf.frontCol);
+            foreach (var (ogcd, (anlock, iscast)) in abilityManager.ogcds) {
+                var isClipping = helper.CheckClip(iscast, ogcd, anlock, gcdTotal, gcdTime);
+                ui.DrawCircSegment(ogcd / gcdTotal, (ogcd + anlock) / gcdTotal, 21f * ui.Scale, isClipping ? conf.clipCol : conf.anLockCol);
+                if (!iscast) ui.DrawCircSegment(ogcd / gcdTotal, (ogcd + 0.04f) / gcdTotal, 23f * ui.Scale, conf.ogcdCol);
+            }
+        }
+
+        public void DrawGCDBar(PluginUI ui) {
+            float gcdTotal = helper.TotalGCD;
+            float gcdTime = helper.lastElapsedGCD;
+
+            if (conf.ShowOnlyGCDRunning && HelperMethods.IsTeleport(DataStore.Action->CastId)) {
+                helper.lastActionTP = true;
+                return;
+            }
+            if (HelperMethods.IsCasting() && DataStore.Action->ElapsedCastTime >= gcdTotal && !HelperMethods.IsTeleport(DataStore.Action->CastId))
+                gcdTime = gcdTotal;
+            if (gcdTotal < 0.1f) return;
+            helper.FlagAlerts(ui);
+            helper.InvokeAlerts((conf.BarWidthRatio + 1) / 2.1f, -0.3f, ui);
+            DrawBarElements(ui, false, helper.shortCastFinished, false, gcdTime / gcdTotal, gcdTime, gcdTotal, gcdTotal);
+
+            // Gonna re-do this, but for now, we flag when we need to carryover from the castbar to the GCDBar
+            // and dump all the crap here to draw on top. 
+            if (helper.shortCastFinished) {
+                string abilityNameOutput = shortCastCachedSpellName;
+                if (helper.queuedAbilityName != " " && conf.CastBarShowQueuedSpell)
+                    abilityNameOutput += " -> " + helper.queuedAbilityName;
+                if (!string.IsNullOrEmpty(abilityNameOutput))
+                    DrawBarText(ui, abilityNameOutput);
+            }
+
+        }
+
+        public void DrawCastBar (PluginUI ui) {
+            float gcdTotal = DataStore.Action->TotalGCD;
+            float castTotal = DataStore.Action->TotalCastTime;
+            float castElapsed = DataStore.Action->ElapsedCastTime;
+            float castbarProgress = castElapsed / castTotal;
+            float castbarEnd = 1f;
+            float slidecastStart = Math.Max((castTotal - conf.SlidecastDelay) / castTotal, 0f);
+            float slidecastEnd = castbarEnd;
+
+            // handle short casts
+            if (gcdTotal > castTotal) {
+                castbarEnd = castTotal / gcdTotal;
+                slidecastStart = Math.Max((castTotal - conf.SlidecastDelay) / gcdTotal, 0f);
+                slidecastEnd = conf.SlideCastFullBar ? 1f : castbarEnd;
+            }
+
+            DrawBarElements(ui, true, gcdTotal > castTotal, gcdTotal < 0.001f, castbarProgress * castbarEnd, slidecastStart, slidecastEnd, castbarEnd);
```

Comment:
The
` gcdTotal < 0.001f`
fails when casting a nonability while still in GCD from a previous ability Instead we could use a function like this in HelperMethods:
```
        public static bool IsNonAbility() {
            return DataStore.ActionManager->CastActionType
                is not ActionType.Action
                and not ActionType.None;
        }
```
But that doesn't fix the mount thing and I don't think it matters for anything else, so unless you can think of an example I think leaving it like that may be okay.